pudb/py3compat.py

```diff
@@ -19,6 +56,47 @@ def execfile(fname, globs, locs=None):
     text_type = unicode  # noqa: F821
     execfile = execfile
 
+    # {{{ container metaclasses
+
+    from abc import ABCMeta
+
+    class PudbCollection:
+        __metaclass__ = ABCMeta
+
+        @classmethod
+        def __subclasshook__(cls, c):
+            if cls is PudbCollection:
+                return all([
+                    any("__contains__" in b.__dict__ for b in c.__mro__),
+                    any("__iter__" in b.__dict__ for b in c.__mro__),
+                ])
+            return NotImplemented
+
+    class PudbSequence:
+        __metaclass__ = ABCMeta
+
+        @classmethod
+        def __subclasshook__(cls, c):
+            if cls is PudbSequence:
+                return all([
+                    any("__getitem__" in b.__dict__ for b in c.__mro__),
+                    any("__iter__" in b.__dict__ for b in c.__mro__),
+                ])
+            return NotImplemented
+
+    class PudbMapping:
+        __metaclass__ = ABCMeta
+
+        @classmethod
+        def __subclasshook__(cls, c):
+            if cls is PudbMapping:
+                return all([
+                    any("__getitem__" in b.__dict__ for b in c.__mro__),
+                    any("__iter__" in b.__dict__ for b in c.__mro__),
+                    any("keys" in b.__dict__ for b in c.__mro__),
+                ])
+            return NotImplemented
+    # }}}
```

Comment:
We can simply drop Py2-specific bits at this point. (`setup.py` says pudb is 3.6+ only, and has for a while.) (And this should probably live in `var_view`.)

Reply:
Right! Sorry I forgot- I'm unforunately stuck in a situation at work where I still have to use Python 2, I'll move this stuff to my branch that maintains py2 support.